# This notebook is an exercise in data analysis.

### The data under analysis is the Kaggle Titanic dataset. Questions will be asked about a what attributes of a person predict their chances of surviving the accident.

In [1]:
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
fields = ['Survived', 'Pclass', 'Sex', 'Age','SibSp',
          'Parch', 'Fare', 'Cabin', 'Embarked']

In [3]:
df = pd.read_csv('./train.csv',usecols=fields)

### The sinking of the Titanic was a deadly event. 

In [4]:
total_people = len(df)

In [5]:
survived = df.loc[df["Survived"]==1]
total_survivors = len(survived)
probability_survived = total_survivors/total_people

In [6]:
perished = df.loc[df["Survived"]==0]
total_perished = len(perished)
probability_perished = total_perished/total_people

In [7]:
print("{:.2f}% survived. {:.2f}% perished".\
      format(probability_survived*100,probability_perished*100))

38.38% survived. 61.62% perished


Survived (%) | Perished (%)
--- | --- |
38.38 | 61.62

It was more likely to perish in the Titanic than to survive.

==> P(Survived) = 38.38%

==> P(Perished) = 61.62%

#### What is the effect of gender on chances of survival?

In [8]:
probability_male = len(df.loc[df["Sex"]==\
                    "male"])/total_people

probability_female = len(df.loc[df["Sex"]==\
                    "female"])/total_people

In [9]:
print("{:.2f}% are women. {:.2f}% are men".\
      format(probability_female*100,probability_male*100))

35.24% are women. 64.76% are men


% Female | % Male
--- | --- |
35.24 | 64.76

You were more likely to come across a male on the Titanic that day.

==> P(Female) = 35.24%

==> P(Male)= 64.76%

In [10]:
female_survivors = len(survived.loc[survived["Sex"]=="female"])
male_survivors = len(survived.loc[survived["Sex"]=="male"])

In [11]:
probability_female_survived = female_survivors/total_survivors
probability_male_survived = male_survivors/total_survivors

print("{:.2f}% of survivors are female. {:.2f}% of survivors are male"\
      .format(probability_female_survived*100,probability_male_survived*100))

68.13% of survivors are female. 31.87% of survivors are male


Gender | survival chance (%)
--- | --- |
Female | 68.13
Male | 31.87

It can be seen that the probability that a survivor is female is 68.13\%.

==> P(Female \| Survived) = 68.13\%

The probability that a survivor is male is 31.87\%.

==> P(Male \| Survived) = 31.87\%

In [12]:
female_perished = len(perished.loc[perished["Sex"]=="female"])
male_perished = len(perished.loc[perished["Sex"]=="male"])

In [13]:
probability_female_perished = female_perished/total_perished
probability_male_perished = male_perished/total_perished

print("{:.2f}% of the deceased are female. {:.2f}% of the \
deceased are male".format(probability_female_perished*100,probability_male_perished*100))

14.75% of the deceased are female. 85.25% of the deceased are male


Gender | perish chance (%)
--- | --- |
Female | 14.75
Male | 85.25

It can be seen that the probability that the person deceased is female is 14.75\%.

==> P(Female \| Deceased) = 14.75\%

The probability that person deceased is male is 85.25\%.

==> P(Male \| Deceased) = 85.25\%

In [14]:
.1475*.6162 + .6813*.3838

0.35237244

In [15]:
(.1475*.6162)/.3524

0.25791572077185015